### PR TITLE
Set up GroupCreateCommand class

### PR DESCRIPTION
**Pull request by lelerer**

### CODE REVIEW

File: src/main/java/seedu/splitlah/command/GroupCreateCommand.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+        ArrayList<Person> personList = convertToListOfPerson();
+
+        boolean isGroupExists = manager.getProfile().hasGroupName(groupName);
+
+        if (isGroupExists) {
+            manager.getUi().printlnMessage(Message.ERROR_GROUPCREATE_DUPLICATE_GROUP_NAME);
+            return;
+        }
+        int newGroupId = manager.getProfile().getNewGroupId();
```
Comment (froststein) [inline src/main/java/seedu/splitlah/command/GroupCreateCommand.java:124]: Perhaps shift this LOC down to line 130, when all the checks have passed. As getNewGroupId is a function that has an effect in the system's variables. It is best to only call when all checks have been passed.

File: src/main/java/seedu/splitlah/data/Group.java [hunk 2]
```diff
@@ -19,18 +19,18 @@
...
+
     /**
      * Constructs a Group object.
      *
-     * @param personList An ArrayList object containing Person objects representing
-     *                   participants of the group.
-     * @param groupName  A String object that represents the group's name.
-     * @param groupId    An integer that uniquely identifies a group.
+     * @param groupName A String object that represents the group's name.
+     * @param groupId An integer that uniquely identifies a group.
```
Comment (froststein) [inline src/main/java/seedu/splitlah/data/Group.java:27]: May want to align all the @param for standardization.

**Review by froststein — CHANGES_REQUESTED** [review]
Code may contain bugs during usage. Need to perform some standardization. Other than that, they are small changes to be made.

File: src/main/java/seedu/splitlah/ui/Message.java [hunk 3]
```diff
@@ -229,8 +229,18 @@
...
+    public static final String ERROR_GROUPCREATE_DUPLICATE_GROUP_NAME =
+            "There is another group with the same name.";
+    public static final String ERROR_GROUPCREATE_DUPLICATE_NAMES =
+            "There are duplicate names in the person list for the group you are trying to create.";
+    public static final String ERROR_GROUPCREATE_DUPLICATE_GROUP_ID =
+             "There is another group with the same Id.";
+    public static final String ASSERT_GROUPCREATE_NAME_DUPLICATE_EXISTS_BUT_NOT_DETECTED =
+             "Name duplicates exists but not detected.";
+
+
```
Comment (warrencxw) [inline src/main/java/seedu/splitlah/ui/Message.java:245]: Consider removing these extra new lines here for standardisation.

File: src/main/java/seedu/splitlah/command/GroupCreateCommand.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import seedu.splitlah.data.Manager;
+import seedu.splitlah.data.Person;
+import seedu.splitlah.exceptions.InvalidFormatException;
+import seedu.splitlah.parser.Parser;
+import seedu.splitlah.ui.Message;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
```
Comment (warrencxw) [inline src/main/java/seedu/splitlah/command/GroupCreateCommand.java:13]: Consider removing one new line here for standardisation.

File: src/main/java/seedu/splitlah/command/GroupCreateCommand.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        "The group was created successfully.\n";
+
+    public static final String[] COMMAND_DELIMITERS = {
+        Parser.NAME_DELIMITER,
+        Parser.PERSON_LIST_DELIMITER
+    };
+
+    private String groupName;
+    private String[] personNames;
+
```
Comment (warrencxw) [inline src/main/java/seedu/splitlah/command/GroupCreateCommand.java:37]: Consider removing this newline here for standardisation

File: src/main/java/seedu/splitlah/data/Group.java [hunk 6]
```diff
@@ -19,18 +19,18 @@
     private static final String PERSON_LIST_HEADER = "Participants";
     private static final int OFFSET = 1;
 
+
```
Comment (warrencxw) [inline src/main/java/seedu/splitlah/data/Group.java:22]: Consider removing this newline here for standardisation.

**Review by warrencxw — APPROVED** [review]
Generally looks good, but many unnecessary new lines have been added which can be removed to standardise with the rest of the code and maintain flow.

File: src/main/java/seedu/splitlah/command/GroupCreateCommand.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            Person newPerson = new Person(name);
+            personList.add(newPerson);
+        }
+        return personList;
+    }
+
+    /**
+     * Checks if String object array of names has duplicated names.
+     *
+     * @return true if it contains duplicates, false otherwise.
```
Comment (ivanaitzliddat) [inline src/main/java/seedu/splitlah/command/GroupCreateCommand.java:67]: Possible to put "false otherwise" on a newline and aligned it with "true if ..."

**Review by ivanaitzliddat — APPROVED** [review]
Looks good overall, left a small comment to be edited for overall standardization

**Pull request merged**